Mage/src/main/java/mage/abilities/effects/common/counter/AddCountersAllEffect.java

```diff
@@ -62,9 +60,9 @@ public boolean apply(Game game, Ability source) {
                     }
 
                     permanent.addCounters(newCounter, source.getControllerId(), source, game);
-                    if (!game.isSimulation()) {
+                    if (!game.isSimulation() && newCounter.getCount() > 0) {
```

Comment:
fyi there's no need to check `isSimulation()` because `informPlayers()` already checks for that